packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -103,11 +96,14 @@ export class BlockList extends Component {
 					ListFooterComponent={ withFooter && this.renderBlockListFooter }
 				/>
 
-				{ renderAppender && blockClientIds.length > 0 &&
-					<BlockListAppender
-						rootClientId={ this.props.rootClientId }
-						renderAppender={ this.props.renderAppender }
-					/>
+				{ renderAppender && blockClientIds.length > 0 && (
+					<View style={ [ ! this.props.hasInnerBlocks && { paddingHorizontal: 12 } ] }>
+						<BlockListAppender
+							rootClientId={ this.props.rootClientId }
+							renderAppender={ this.props.renderAppender }
+						/>
+					</View>
+				)
```

Comment:
Also here I feel that maybe there is better way to adjust appender margin inside the selected group instead of wrapping in into `View`.

`renderAppender` render prop is passed from `edit.native.js` which know if group is empty or not but then we need to inject this prop to `ButtonBlockAppender`

Alternatively we can add `withSelect` to move above logic to `ButtonBlockAppender` component which will apply the desire padding around it

Reply:
I don't understand what's going on here. By the name `hasInnerBlocks` I would expect that to be equivalent to `blockClientIds.length > 0`, which we check just above, and this condition would never be true. However, `hasInnerBlocks` is defined as `selectedBlock && selectedBlock.innerBlocks.length !== 0`, which translates to "there is a selected block which has inner blocks".

I don't understand why this is being controlled by the selected block's children and not the current block.

Reply:
Ok, you're right that the `blockClientIds.length >0 ` already includes the case where the group has any children inside so below snippet will be enough to apply padding that I need.

<image src="https://user-images.githubusercontent.com/21242757/69624183-7da4e300-1044-11ea-8ab1-b389253dc7ed.png" width="400" />

But how about wrapping `BlockListAppender` with `View` in here. Is it ok for you ?

The flow behind `renderAppender` is as follow. It is passed from `group/edit.native.js` as a reference to `InnerBlocks.ButtonBlockAppender`. Then inside the `ButtonBlockAppender` appender is rendered inside `renderToggle` of `Inserter`.

Reply:
So if the padding is added for every appended, why don’t we make it a margin in the appender’s styles? Is there another case where the appended is used?

Reply:
We can do this in `ButtonBlockAppender` : 
<image src="https://user-images.githubusercontent.com/21242757/69634447-7f75a300-1052-11ea-9b37-2874dfe65c88.png" width="350" />


But then we will break the margins when we select empty group. However refactor this component to this will cover that case but I'm not sure is proper approach as well.

<details>

```
function ButtonBlockAppender( { rootClientId, hasInnerBlocks, getStylesFromColorScheme } ) {
	const appenderStyle = getStylesFromColorScheme( styles.appender, styles.appenderDark );
	const addBlockButtonStyle = getStylesFromColorScheme( styles.addBlockButton, styles.addBlockButtonDark );

	return (
		<>
			<Inserter
				rootClientId={ rootClientId }
				renderToggle={ ( { onToggle, disabled, isOpen } ) => (
					<View style={ hasInnerBlocks && styles.paddingToContent } >
						<Button
							onClick={ onToggle }
							aria-expanded={ isOpen }
							disabled={ disabled }
							fixedRatio={ false }
						>
							<View style={ [ appenderStyle, { flex: 1 } ] }>
								<Dashicon
									icon="plus-alt"
									style={ addBlockButtonStyle }
									color={ addBlockButtonStyle.color }
									size={ addBlockButtonStyle.size }
								/>
							</View>
						</Button>
					</View>
				) }
				isAppender
			/>
		</>
	);
}

/**
 * @see https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/button-block-appender/README.md
 */
export default compose( [
	withSelect( ( select, { rootClientId } ) => {
		const {
			getBlockCount,
		} = select( 'core/block-editor' );

		return {
			hasInnerBlocks: !! getBlockCount( rootClientId ),
		};
	} ),
	withPreferredColorScheme,
] )( ButtonBlockAppender );
```

</details>

Reply:
After slack discussion we decide to keep the original solution which is apply `View` on the top of `BlockListAppender` and get style padding from .SCSS stylesheet